### PR TITLE
Remove use of the deprecated methods

### PR DESCRIPTION
**Pull request by mikeger**
- Initial PR to check the CI compilation

### CODE REVIEW

File: ConfigurationSets/Warnings.xcconfig [hunk 1]
```diff
@@ -16,7 +16,7 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 // 
 
-
+GCC_TREAT_WARNINGS_AS_ERRORS = YES
```
Comment (vytis) [inline ConfigurationSets/Warnings.xcconfig]: 🥇

**Comment by vytis** [comment]
I guess we will still have the "unsafe in extension" warnings? Will that stop compilation as well?

File: ConfigurationSets/Warnings.xcconfig (hunk 1 above)
Comment (mikeger) [inline ConfigurationSets/Warnings.xcconfig]: It was on for Objective-C before, just moved it to `xcconfig` :)

Comment (vytis) [inline ConfigurationSets/Warnings.xcconfig]: Ah I see, not very ground-breaking then:)

**Comment by mikeger** [comment]
@vytis the warnings about frameworks have to be fixed in the frameworks, this would be the next step. Those are linker warnings and they are not treated as errors (however they should be).

**Comment by vytis** [comment]
Yeah, this could be done separately, fixing UI project warnings is more important anyway.

**Comment by mikeger** [comment]
This one is good to go now.

**Pull request merged**